soccer/gameplay/plays/offense/adaptive_formation.py

```diff
@@ -148,11 +153,18 @@ def __init__(self):
 
     @classmethod
     def score(cls):
-        if (not main.game_state().is_playing()):
-            return float("inf")
-        if len(main.our_robots()) < 5:
-            return float("inf")
-        return 8
+        score = super().score()
```

Comment:
What is this doing now???

Reply:
This calls the score function in standard play, if it returns none then you are supposed to use your own score function, if not it gives you the starting point for your score based on the situation list you have made for the class. I'll add more documentation before merge.

Reply:
Honestly, I would rather have this hook into the system in a manner other than the `score` function - if you do it this way you have to properly implement the super-call.

I would _prefer_ the play selection explicitly check against `is_applicable` for each play, and only check `score` if `is_applicable` returns true.

Reply:
Just in general something to keep in mind, the less boilerplate that needs to be added to new plays the better. It's much easier to mess with and see what a play is doing if you don't have to skip over tons of repeated stuff. If something is really simple, it should look really simple functionality wise.